merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -713,16 +717,24 @@ public InProgress continueWith(final Task newState) {
     }
 
     /** The task has completed its primary operation, but has unfinished children. */
-    record AwaitingChildren(Duration startOffset, Duration endOffset, LinkedList<TaskId> remainingChildren)
-        implements ExecutionState
+    record AwaitingChildren(
+        Duration startOffset,
+        Duration endOffset,
+        LinkedList<TaskId> remainingChildren,
+        TaskReturnValue<?> returnValue
```

Comment:
I know it's totally non-obvious, but the ordering the parameters here is meant to separate things into two groups: the "fixed" data that's merely been logged, and the "active" data that's modifed as the task state machine moves along. There's a lot more that could be done to make this obvious, but here I'd simply recommend making sure that `remainingChildren` is last.

Reply:
Reordered :+1: